Goon-style blob placement

ChangelingRain opened the pull request:
:cl: Joan
rscdel: Blob gamemode blobs will no longer burst from people.
rscadd: Instead, the blob gamemode spawns overminds, which can place their blob core underneath them in an unobserved area.
rscadd: The overminds can, until they have placed a blob core, move to any non-space, non-shuttle tile.
rscadd: If the overminds fail to place a blob core within a time limit, it will be automatically placed for them at a random blob spawnpoint.
tweak: The time before the overmind automatically places the core is slightly lower than the previous average time it'd take a blob to burst.
/:cl:

None of the code is taken from goon, by the way, I wrote all of it.
Fixes #15296 (by removing it)
Fixes #16056 (by preventing movement there)

ChangelingRain committed:
Goon-style blob placement

lzimann commented:
.>no more sabotage before as human

not a fan of it

MMMiracles commented:
Being able to sabotage shit beforehand if you were fast enough was interesting but in the end it just always ended up being "release singulo/destroy RnD/space armory".

Also you should probably note that you decreased the amount of times blobs have before automatic placement in the changelog.

ChangelingRain committed:
ah,,,,,

ChangelingRain commented:
The average time is slightly lower, yeah, but I wasn't sure it was really notable, I guess I will.

KorPhaeron commented:
Human sabotage was more or less the only variance between blob rounds.

Incoming5643 commented:
I misread the title and thought you were porting the blob smoothing goon has. Then I got disappointed.

I can't say I'm a fan of cutting out the shenanigans preburst phase either.

AdamElTablawy commented:
>one blob places core in armory and eats all the guns

>the other one puts it in the AI sat/tcomms

>every round

ChangelingRain commented:
Blobs can't place in the AI core because they can't reach it(and even if they could, there's an AI in there and it'd prevent placement)
Similarly, they can't place in the armory, because the warden can see them and it'll prevent placement.

Shadowlight213 commented:
New meta, hos and scientists stand watch over armory and rnd for the auto place time to deter blobs

ChangelingRain committed:
Thoroughness in placement checking;
can't place on dense turfs
can place on dense objects
can't place if there's someone in a large view()

Zilenan91 commented:
Perma blob will happen every round now

Though yeah it's probably a good thing that this happens. It felt like a cheap, hollow victory when a HoP blob burst with the entire armory and a subverted AI while I was a fellow blob. Nobody could fight us.

Shadowlight213 commented:
Subverting the AI as blob? Absolutely brilliant planning.

YakumoChen commented:
no way to prevent the xray spam every round now, gg blob balance

newfren commented:
If you want to prevent Xray spam then stop the minerals from getting back. Still not sure how I feel about this.

phil235 commented on code/game/gamemodes/blob/overmind.dm:
```diff
@@ -18,8 +18,22 @@
+	if(placed) //we already have a core!
+		manualplace_min_time = 0
+		autoplace_max_time = 0
+		placed = 1
```
if(placed) placed = 1 ?

ChangelingRain replied:
whoop yeah I should fix that

ChangelingRain committed:
fixes

phil235 commented on code/game/gamemodes/blob/blobs/core.dm:
```diff
@@ -15,16 +15,16 @@
 
 
-/obj/effect/blob/core/New(loc, var/h = 200, var/client/new_overmind = null, var/new_rate = 2)
+/obj/effect/blob/core/New(loc, var/client/new_overmind = null, var/new_rate = 2, var/placed = 0)
```
don't need "var/" for the arguments.

ChangelingRain committed:
this is old code

Iamgoofball commented:
yeah you can stop xray spam if you pop in the mining dock/nearby R&D

PKPenguin321 commented:
do we really want blobs popping in mining/R&D every round tho
also does the blob overmind get night vision? it should so that it can still pop in maint

ChangelingRain commented:
The meta is already 'disabling R&D and or mining' so this isn't that big of a change on that side.
And yeah it already has nightvision.

PKPenguin321 commented:
disabling them != popping in them

AdamElTablawy commented:
>blob pops near mining
>minerals never come
>gg

>blob pops in tcomms
>other guy distractionpops in R&D

>gg

lzimann commented:
:-1:

Aloraydrel commented:
Rip blob being nerfed every pr

Incoming5643 commented:
Why
